### PR TITLE
RoadmapLevel component

### PR DESCRIPTION
**Pull request by bernardodiasc**
https://github.com/x-team/auto-ui/issues/8

### CODE REVIEW

**Comment by bernardodiasc** [comment]
Currently having an error on using other UI components:

> TypeError: Failed to construct 'Text': Please use the 'new' operator, this DOM object constructor cannot be called as a function.

I tried the `new` operator, didn't worked, any suggestion? 😢 

/cc @joshwnj

File: src/components/RoadmapLevel.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+type Props = {
+  icon: string,
+  heading: Element<*>|string,
+  level: number,
+  body: Element<*>|string,
+  isActive: boolean,
+}
+
+const Root = elem.section(cmz(``))
```
Comment (joshwnj) [inline src/components/RoadmapLevel.js]: better to just do `elem.section()` if there's no styles.

**Comment by bernardodiasc** [comment]
This PR is ready to be reviewed.

**Note that the `Button` component is still missing (https://github.com/x-team/auto-ui/issues/14)**


Current state:

screenshot-localhost-6060 2017-11-12 11-21-50-082

**Comment by bernardodiasc** [comment]
Another note: this branch contain commits from #29 and #31 (and some minor extra changes on `SvgIcon` and `Text`). And both other PRs will be automatically merged if this one gets merged first.

**Comment by joshwnj** [comment]
Ok let's wait until the dependencies are merged before reviewing this 👍

**Comment by joshwnj** [comment]
@bernardodiasc could you please do a cleanup of this one now that the dependencies are merged? Then we should be good to go :)

**Comment by bernardodiasc** [comment]
hey @joshwnj this branch is merged from latest of master now. It contain a lot snap files since it touched Text component too.

I forgot to fix the flow error before, but it's late night here, I can do the fix tomorrow morning.

```
$ npm run flow

> auto-ui@1.0.0 flow /Users/bernardo/Sites/auto-ui
> flow check src/

Error: src/components/RoadmapLevel.js:85
           v-------
 85:       <SvgIcon
 86:         icon={icon}
 87:       />,
           -^ props of React element `SvgIcon`. This type is incompatible with
161: class SvgIcon extends PureComponent<Props> {
                                         ^^^^^ object type. See: src/components/SvgIcon.js:161
  Property `icon` is incompatible:
     86:         icon={icon}
                       ^^^^ string. This type is incompatible with
      7:   icon: 'head' | 'webcam' | 'message' | 'terminal' | 'diamond' | 'talking' | 'trophy',
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ string enum. See: src/components/SvgIcon.js:7

Error: src/components/RoadmapLevel.js:85
           v-------
 85:       <SvgIcon
 86:         icon={icon}
 87:       />,
           -^ props of React element `SvgIcon`. This type is incompatible with
161: class SvgIcon extends PureComponent<Props> {
                                         ^^^^^ object type. See: src/components/SvgIcon.js:161
  Property `icon` is incompatible:
    163:     icon: '',
                   ^^ string. This type is incompatible with. See: src/components/SvgIcon.js:163
      7:   icon: 'head' | 'webcam' | 'message' | 'terminal' | 'diamond' | 'talking' | 'trophy',
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ string enum. See: src/components/SvgIcon.js:7
```

**Comment by joshwnj** [comment]
@bernardodiasc thanks! All good, I can fix now.

**Comment by joshwnj** [comment]
These flow warnings can appear cryptic at first, but I find with a bit of time they are more meaningful.

```
Property `icon` is incompatible:
    163:     icon: '',
                   ^^ string. This type is incompatible with. See: src/components/SvgIcon.js:163
      7:   icon: 'head' | 'webcam' | 'message' | 'terminal' | 'diamond' | 'talking' | 'trophy',
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ string enum. See: src/components/SvgIcon.js:7
```

This warning says _"you gave `icon` a default value of an empty string, but the type definition says it must be one of these strings"_.

Makes sense, right? We don't want a default of `icon: ''`, because an `SvgIcon` with no icon is meaningless. It only makes sense if we use the name of a valid icon.

I've removed the default prop value, as `undefined` is more meaningful than an empty string in this case.

The other part of the error was that `RoadmapLevel` defined its `Props` type like this:

```
type Props = {
  icon: string,
```

So flowtype complains, because it says _"you say that `icon` is a string, but I know that `SvgIcon` expects a very particular string, not just any old string"_.

So to fix this I've exported `type Icon` from `SvgIcon`, and updated `RoadmapLevel` to

```
import type { Icon } from './SvgIcon'

type Props = {
  icon: ?Icon,
```

Let me know if any confusion remains around this, or if it's all clear now. Thanks!

**Pull request merged**